tests/kernels/moe/modular_kernel_tools/mk_objects.py

```diff
@@ -449,34 +482,34 @@ def make_fused_experts(
 
     torch.set_printoptions(threshold=0, edgeitems=0, linewidth=10000)
 
-    if fused_experts_type == BatchedDeepGemmExperts:
+    if fused_experts_type.__name__ == "BatchedDeepGemmExperts":
         kwargs = batch_kwargs | quant_kwargs
         print(f"Making BatchedDeepGemmExperts {kwargs} ...")
         experts = BatchedDeepGemmExperts(**kwargs)
-    elif fused_experts_type == BatchedTritonExperts:
+    elif fused_experts_type.__name__ == "BatchedTritonExperts":
         kwargs = batch_kwargs | quant_kwargs
         print(f"Making BatchedTritonExperts {kwargs} ...")
         experts = BatchedTritonExperts(**kwargs)
-    elif fused_experts_type == BatchedTritonOrDeepGemmExperts:
+    elif fused_experts_type.__name__ == "BatchedTritonOrDeepGemmExperts":
         kwargs = batch_kwargs | quant_kwargs | deepgemm_kwargs
         print(f"Making BatchedTritonOrDeepGemmExperts {kwargs} ...")
         experts = BatchedTritonOrDeepGemmExperts(**kwargs)
-    elif fused_experts_type == DeepGemmExperts:
+    elif fused_experts_type.__name__ == "DeepGemmExperts":
         print(f"Making DeepGemmExperts {quant_config} ...")
         experts = DeepGemmExperts(quant_config)
-    elif fused_experts_type == TritonExperts:
+    elif fused_experts_type.__name__ == "TritonExperts":
         kwargs = quant_kwargs
         print(f"Making TritonExperts {kwargs} ...")
         experts = TritonExperts(**kwargs)
-    elif fused_experts_type == TritonOrDeepGemmExperts:
+    elif fused_experts_type.__name__ == "TritonOrDeepGemmExperts":
         kwargs = quant_kwargs | deepgemm_kwargs
         print(f"Making TritonOrDeepGemmExperts {kwargs} ...")
         experts = TritonOrDeepGemmExperts(**kwargs)
-    elif fused_experts_type == NaiveBatchedExperts:
+    elif fused_experts_type.__name__ == "NaiveBatchedExperts":
         kwargs = batch_kwargs | quant_kwargs
         print(f"Making NaiveBatchedExperts {kwargs} ...")
         experts = NaiveBatchedExperts(**kwargs)
-    elif fused_experts_type == CutlassExpertsFp8:
+    elif cutlass_fp8_supported() and fused_experts_type.__name__ == "CutlassExpertsFp8":
```

Comment:
nit: I think it would be better to make the support checks asserts under each branch rather than making the construction conditional on them.

Reply:
done